lib/tapioca/compilers/dsl/active_record_associations.rb

```diff
@@ -0,0 +1,285 @@
+              ],
+              return_type: association_type
+            )
+          end
+        end
+
+        sig do
+          params(
+            klass: Parlour::RbiGenerator::Namespace,
+            constant: T.class_of(ActiveRecord::Base),
+            association_name: T.any(String, Symbol),
+            reflection: ReflectionType
+          ).void
+        end
+        def populate_collection_assoc_getter_setter(klass, constant, association_name, reflection)
+          association_class = type_for(constant, reflection)
+          relation_class = relation_type_for(constant, reflection)
+
+          create_method(
+            klass,
+            association_name.to_s,
+            return_type: relation_class,
+          )
+          create_method(
+            klass,
+            "#{association_name}=",
+            parameters: [
+              Parlour::RbiGenerator::Parameter.new("value", type: "T::Enumerable[#{association_class}]"),
+            ],
+            return_type: nil,
+          )
+          create_method(
+            klass,
+            "#{association_name.to_s.singularize}_ids",
+            return_type: "T::Array[T.untyped]"
+          )
+          create_method(
+            klass,
+            "#{association_name.to_s.singularize}_ids=",
+            parameters: [
+              Parlour::RbiGenerator::Parameter.new("ids", type: "T::Array[T.untyped]"),
+            ],
+            return_type: "T::Array[T.untyped]"
+          )
+        end
+
+        sig do
+          params(
+            constant: T.class_of(ActiveRecord::Base),
+            reflection: ReflectionType
+          ).returns(T::Boolean)
+        end
+        def belongs_to_and_required?(constant, reflection)
+          return false unless constant.table_exists?
+          return false unless reflection.belongs_to?
+          column_definition = constant.columns_hash[reflection.foreign_key.to_s]
+
+          !column_definition.nil? && !column_definition.null
+        end
+
+        sig do
+          params(
+            constant: T.class_of(ActiveRecord::Base),
+            reflection: ReflectionType
+          ).returns(String)
+        end
+        def type_for(constant, reflection)
+          return "T.untyped" if !constant.table_exists? || polymorphic_association?(reflection)
+
+          "::#{reflection.klass.name}"
+        end
+
+        sig do
+          params(
+            constant: T.class_of(ActiveRecord::Base),
+            reflection: ReflectionType
+          ).returns(String)
+        end
+        def relation_type_for(constant, reflection)
+          "ActiveRecord::Associations::CollectionProxy" if !constant.table_exists? ||
+                                                            polymorphic_association?(reflection)
+
+          # Change to: "::#{reflection.klass.name}::ActiveRecord_Associations_CollectionProxy"
+          "::ActiveRecord::Associations::CollectionProxy[#{reflection.klass.name}]"
```

Comment:
Hmm, this is a problem, since there is no actual module `ActiveRecord::Associations::CollectionProxy` which has a generic parameter. That is just an RBI shim file we have in core that defines that module and we don't have that for any other class. I guess we can merge this as is and then fix this when we do the refactor to replace it with `Model::ActiveRecord::Associations::CollectionProxy`.